### ISSUE TITLE
shortest_angular_distance_with_limits bug

### ISSUE THREAD

**Issue by davetcoleman** [ros/angles#2]
I believe I've found a bug in angles... if I call:

```
from = 1.04700021
to = 1.04700027
left_limit = 1.047
right_limit = -2.147

result = shortest_angular_distance_with_limits(from, to, left_limit, right_limit, shortest_angle)
```
I get
```
shortest_angle = -6.28318525
result = 0
```

Where the shortest_angle should be a small decimal number. So I guess its calculating the angle all the way around the unit circle, instead of the small sliver of distance between the two positions (my application is the effort-based position controller in ros_control).

**Comment by AtsushiSakai** [comment]
I think this issue has been already fixed.

    double from = 1.04700021;
    double to = 1.04700027;
    double  left_limit = 1.047;
    double  right_limit = -2.147;

    double shortest_angle;
    double result = angles::shortest_angular_distance_with_limits(from, to, left_limit, right_limit, shortest_angle);
    cout<<"shortest_angle is "<<shortest_angle<<endl;
    cout<<"result is "<<result<<endl;

then, I get 

    shortest_angle is 6e-08
    result is 1

**Comment by tfoote** [comment]
Thanks, yes. I believe this was resolved by fa862364842c614cbaab1587ea4e5b19b7cbc536
